strongbox-web-core/src/main/java/org/carlspring/strongbox/controller/UserController.java

```diff
@@ -227,12 +219,13 @@ ResponseEntity delete(@ApiParam(value = "The name of the user") @PathVariable St
     @ApiResponses(value = { @ApiResponse(code = 200, message = "The security token was generated."),
                             @ApiResponse(code = 500, message = "An error occurred.") })
     @PreAuthorize("hasAuthority('UPDATE_USER')")
-    @RequestMapping(value = "user/{name}/generate-security-token", method = RequestMethod.GET)
-    public ResponseEntity generateSecurityToken(@ApiParam(value = "The name of the user") @PathVariable String name)
+    @RequestMapping(value = "user/{userName}/generate-security-token", method = RequestMethod.GET)
+    public ResponseEntity generateSecurityToken(@ApiParam(value = "The name of the user") @PathVariable String userName)
         throws JoseException
 
     {
-        User user = userService.findByUserName(name);
+        // XXX: WTF??? Without this we have a 'User' proxy with empty 'id'
```

Comment:
You have to detach entity from database all the time you need particular property of it instead of proxied one. So, use detachAll() method before accessing properties like 'id'.